README.md

```diff
@@ -70,3 +70,118 @@ The official deployed contract addresses
 - Factory: `ct_NhbxN8wg8NLkGuzwRNDQhMDKSKBwDAQgxQawK7tkigi2aC7i9`
 - Router  `ct_MLXQEP12MBn99HL6WDaiTqDbG4bJQ3Q9Bzr57oLfvEkghvpFb`
 - Wrapped AE: `ct_JDp175ruWd7mQggeHewSLS1PFXt9AzThCDaFedxon8mF8xTRF`
+
+## Using the DEX contracts in your application
+
+### Trading pair swap price calculation
+To calculate the price of your token (`token_a`) in terms of another token, such as Wrapped Aethernity (`token_b`), 
+you can use the DEX contracts to fetch the necessary information. 
+The price calculation for a trading pair `token_a`/`token_b` involves the following steps:
+
+#### 1. Get trading pair
+Firstly, obtain the trading pair contract address from the DEX factory using the `get_pair` entrypoint. 
+You need the addresses of `token_a` and `token_b`. 
+The function returns the address of the trading pair’s liquidity pool contract:
+```
+AedexV2Factory.get_pair(token_a: IAEX9Minimal, token_b: IAEX9Minimal): option(IAedexV2Pair)
+```
+
+#### 2. Get reserves for pair
+Use the `get_reserves` function of the pair’s contract to obtain the current reserves for `token_a` and `token_b`:
+```
+AedexV2Pair.get_reserves(): { reserve_token_a: int, reserve_token_b:, block_timestamp_last: int }
+```
+
+#### 3. Calculate price for trading pair
+Ensure you know the decimals for `token_a` and `token_b`. The decimals can be retrieved from the token contract's `meta_info`.
+If the tokens' decimals vary, normalize the reserves by adjusting for the token decimals:
+- `reserve_token_a_normalized = reserve_token_a * 10^(-decimals_token_a)`
+- `reserve_token_b_normalized = reserve_token_b * 10^(-decimals_token_b)`
+
+After normalization, calculate the trading prices as follows:
+- Price `p` of `token_a` in `token_b`: 
+  - `p = reserve_token_b_normalized / reserve_token_a_normalized`
+  - &rarr; for `x token_a` you get `y token_b`, where `y = p * x`
+- Remember to also account for the 0.3% trading fee, which affects the final received amount:
+  - `final_amount = y - (y * 0.003)`
+
+In Sophia the price calculation looks like this:
+```Sophia
+entrypoint calculate_price(factory: IAedexV2Factory, token_a: IAEX9Minimal, token_b: IAEX9Minimal) = 
+
+  let token_a_decimals = token_a.meta_info().decimals
+  let token_b_decimals = token_b.meta_info().decimals
+
+  let precision = 10^token_b_decimals
+
+  let pair_option = factory.get_pair(token_a, token_b)
+  let reserves = switch(pair_option)
+    None => abort("No pair found.")
+    Some(pair) => pair.get_reserves()
+
+  // Sophia does not have float numbers so we normalize with 10^decimals instead of 10^-decimals
+  let reserve_token_a_normalized = reserves.reserve0 * 10^token_a_decimals
+  let reserve_token_b_normalized = reserves.reserve1 * 10^token_b_decimals
```

Comment:
das kommt mir irgendwie komisch vor, die reserves sind doch schon in 10^decimals notation, warum machen wir das hier nochmal?